src/plugins/score-plugin-fx/Fx/FactorOracle2.hpp

```diff
@@ -0,0 +1,425 @@
+#pragma once
+#include <Engine/Node/PdNode.hpp>
+
+#include <ossia/detail/hash_map.hpp>
+#include <ossia/detail/math.hpp>
+
+#include <random>
+#if !defined(NDEBUG) && !defined(_MSC_VER) && !defined(__clang__)
+#include <debug/vector>
+#define debug_vector_t __gnu_debug::vector
+#else
+#define debug_vector_t std::vector
+#endif
+
+/*! \file FactorOracle.h
+    \brief A file that contains the definitions of the classes needed for the creation of a Factor Oracle.
+
+    Three main classes: FactorOracle, State and SingleTransition.
+*/
+/*! \fn void AddLetter(FactorOracle& States, vector <vector<int>> &T, int i, string word)
+    \brief Adds new transitions from state i-1 to state i.
+    \param T A vector where each position has all the suffix transitions directed to each state.
+    \param i The integer of the current state.
+    \param word The input string.
+
+*/
+/*! \fn int LengthCommonSuffix(FactorOracle& States, int phi_one, int phi_two)
+    \brief Finds the length of a common suffix ending at the position phi_one and phi_two by traversing the suffix links.
+    \param phi_one The position of the state.
```

Comment:
can you add a link to the paper in the comment ? so that we know what are phi1 / phi2